field/src/field.rs

```diff
@@ -149,9 +149,10 @@ pub trait Field:
         }
     }
 
+    // Default naive square and multiply implementation for powers.
     #[must_use]
     #[inline]
-    fn exp_u64(&self, power: u64) -> Self {
+    fn exp_u64_default(&self, power: u64) -> Self {
```

Comment:
nit: consider making it a global function to emphasize that it can't/shouldn't be overridden